dashboard/test/helpers/delete_accounts_helper_test.rb

```diff
@@ -704,8 +705,165 @@ class DeleteAccountsHelperTest < ActionView::TestCase
+    student = create :student
+    with_channel_for student do |channel_id, storage_id|
+      refute_nil storage_apps.where(id: channel_id).first[:value]
+      storage_apps.where(storage_id: storage_id).each do |app|
+        refute_nil app[:value]
+      end
+
+      purge_user student
+
+      assert_nil storage_apps.where(id: channel_id).first[:value]
+      storage_apps.where(storage_id: storage_id).each do |app|
+        assert_nil app[:value]
+      end
+    end
+  end
+
+  test "clears 'updated_ip' all of a user's projects" do
+    storage_apps = PEGASUS_DB[:storage_apps]
+    student = create :student
+    with_channel_for student do |channel_id, storage_id|
+      refute_empty storage_apps.where(id: channel_id).first[:updated_ip]
+      storage_apps.where(storage_id: storage_id).each do |app|
+        refute_empty app[:updated_ip]
+      end
+
+      purge_user student
+
+      assert_empty storage_apps.where(id: channel_id).first[:updated_ip]
+      storage_apps.where(storage_id: storage_id).each do |app|
+        assert_empty app[:updated_ip]
+      end
+    end
+  end
+
+  #
+  # Testing our utilities
+  #
+
+  test 'with_channel_for owns channel' do
+    table = PEGASUS_DB[:storage_apps]
+    student = create :student
+
+    with_storage_id_for student do |storage_id|
+      assert_empty table.where(storage_id: storage_id)
+
+      with_channel_for student do |channel_id|
+        assert_equal channel_id, table.where(storage_id: storage_id).first[:id]
+      end
+
+      assert_empty table.where(storage_id: storage_id)
+    end
+  end
+
+  test 'with_storage_id_for owns id if it does not exist' do
+    table = PEGASUS_DB[:user_storage_ids]
+    student = create :student
+
+    assert_empty table.where(user_id: student.id)
+
+    with_storage_id_for student do |storage_id|
+      assert_equal storage_id, table.where(user_id: student.id).first[:id]
+    end
+
+    assert_empty table.where(user_id: student.id)
+  end
+
+  test 'with_storage_id_for does not own id if it does exist' do
+    table = PEGASUS_DB[:user_storage_ids]
+    student = create :student
+    assert_empty table.where(user_id: student.id)
+
+    table.insert(user_id: student.id)
+
+    refute_empty table.where(user_id: student.id)
+
+    with_storage_id_for student do |storage_id|
+      assert_equal storage_id, table.where(user_id: student.id).first[:id]
+    end
+
+    refute_empty table.where(user_id: student.id)
+
+    table.where(user_id: student.id).delete
+    assert_empty table.where(user_id: student.id)
+  end
+
   private
 
+  def with_channel_for(owner)
+    table = PEGASUS_DB[:storage_apps]
+    channels_before = table.count
+    with_storage_id_for owner do |storage_id|
+      encrypted_channel_id = StorageApps.new(storage_id).create({projectType: 'applab'}, ip: 123)
+      _, id = storage_decrypt_channel_id encrypted_channel_id
+      yield id, storage_id
+    ensure
+      table.where(storage_id: storage_id).delete
```

Comment:
Agreed that I should delete by `id` (nice catch!).  I'd like to keep the `ensure` relevant, not just to the `yield`, but to the line that creates the channel, since that's what I'm cleaning up.

I'm not worried about the `yield` silently failing to execute when there's some error creating the channel, because `ensure` doesn't swallow exceptions.  Checked in irb:
```
2.5.0 :007 > def ensure_does_not_swallow_exception
2.5.0 :008?>   raise 'Intentional failure'
2.5.0 :009?>   ensure
2.5.0 :010 >     puts "Ensure block ran"
2.5.0 :011?>   end
 => :ensure_does_not_swallow_exception 
2.5.0 :012 > ensure_does_not_swallow_exception
Ensure block ran
Traceback (most recent call last):
        3: from /home/brad/.rvm/rubies/ruby-2.5.0/bin/irb:11:in `<main>'
        2: from (irb):12
        1: from (irb):8:in `ensure_does_not_swallow_exception'
RuntimeError (Intentional failure)

```